manufacturing-ai-suite/industrial-edge-insights-vision/docs/pallet-defect-detection/how-to-integrate-balluff-sdk.md

```diff
@@ -0,0 +1,256 @@
+
+# For Ubuntu 22 with intel/dlstreamer-pipeline-server:3.1.0-ubuntu22, uncomment the line below and comment the above line
+# RUN apt-get update && apt-get install -y libwxgtk-webview3.0-gtk3-dev
+
+RUN mkdir /home/pipeline-server/Balluff_Impact_Acquire_V3
+RUN wget https://assets-2.balluff.com/mvIMPACT_Acquire/3.0.0/install_mvGenTL_Acquire.sh -P /home/pipeline-server/Balluff_Impact_Acquire_V3
+RUN wget https://assets-2.balluff.com/mvIMPACT_Acquire/3.0.0/mvGenTL_Acquire-x86_64_ABI2-3.0.0.tgz -P /home/pipeline-server/Balluff_Impact_Acquire_V3
+RUN cd Balluff_Impact_Acquire_V3 && ./install_mvGenTL_Acquire.sh -u3v -gev -u
+
+ENV MVIMPACT_ACQUIRE_DIR="/opt/mvIMPACT_Acquire" \
+    MVIMPACT_ACQUIRE_DATA_DIR="/opt/mvIMPACT_Acquire/data" \
+    GENICAM_ROOT="/opt/mvIMPACT_Acquire/runtime" \
+    MVIMPACT_ACQUIRE_FAVOUR_SYSTEMS_LIBUSB="1"
+
+USER intelmicroserviceuser
+```
+
+---
+
+### Step 3: Build the Docker Image
+
+Run the following command to build the image:
+
+```bash
+docker build -t intel/dlstreamer-pipeline-server:3.1.0-ubuntu24-gencamsrc-balluff -f BalluffDockerfile .
+```
+
+This command builds your Docker image using the steps defined above.
+
+---
+
+### Step 4: Verify the Image
+
+After the build completes, update .env and start the container:
+
+> update .env DLSTREAMER_PIPELINE_SERVER_IMAGE=intel/dlstreamer-pipeline-server:3.1.0-ubuntu24-gencamsrc-balluff
+
+```bash
+docker compose up -d
+```
+
+---
+
+### Step 5: Testing
+
+Note down serial of the balluff camera and up `<balluff-camera-serial>` in the following command
+```bash
+docker exec -it dlstreamer-pipeline-server bash
+$ export GENICAM_GENTL64_PATH=/opt/Impact_Acquire/lib/x86_64
+$ export GST_PLUGIN_PATH=$GST_PLUGIN_PATH:/usr/lib/x86_64-linux-gnu/gstreamer-1.0:/usr/local/lib/gstreamer-1.0
+$ gst-launch-1.0 gencamsrc serial=<balluff-camera-serial> pixel-format=bayerrggb name=source ! bayer2rgb ! videoscale ! video/x-raw, width=1920,height=1080 ! videoconvert ! queue ! jpegenc ! avimux ! filesink location=/tmp/gencam_balluff_output.avi
+```
+
+Verify that the /tmp/gencam_balluff_output.avi has the captured content
+
+---
+
+## Deploying the Pallet Defect Detection (PDD) Application Using live camera
+
+This guide provides detailed, step-by-step instructions for setting up and deploying the **Pallet Defect Detection (PDD)** pipeline to use the **Balluff** or **Basler** camera connected over USB or GigE.  
+It covers environment setup, configuration updates, and validation steps to ensure a successful deployment.
+
+---
+
+### Step 1: Set Up the Environment
+
+```bash
+git clone https://github.com/open-edge-platform/edge-ai-suites.git
+cd edge-ai-suites/manufacturing-ai-suite/industrial-edge-insights-vision
+cp .env_pallet_defect_detection .env
+```
+
+---
+
+### Step 2: Configure the .env File
+
+Open the `.env` file and update the following variables:
+
+```bash
+HOST_IP=
+DLSTREAMER_PIPELINE_SERVER_IMAGE=intel/dlstreamer-pipeline-server:3.1.0-ubuntu24-gencamsrc-balluff
```

Comment:
We dont have to include/add other variables from env, it may confuse if all 6-7 ENVs are required to run GIGe camera.. Only this 1-line should be enough and we can say update/add this line/variable.